maestro/audit.py

```diff
@@ -351,6 +352,53 @@ def from_config(cfg):
             cfg.get('timeout', WebHookAuditor.DEFAULT_TIMEOUT))
 
 
+class ExecuteScriptAuditor(BaseAuditor):
+    """Auditor that executes scripts with maestro action information."""
+
+    def __init__(self, script, args, level):
+        super(ExecuteScriptAuditor, self).__init__(level)
+        if not script:
+            raise exceptions.InvalidAuditorConfigurationException(
+                'Missing script to execute!')
+
+        self._script = script
+        self._args = args
+
+    def _format_container_dict(self, what):
+        d = {}
+        for item in what:
+            x = dict(ship=item.ship.ip, service=item.service.name)
+            d[item.name] = x
+        return json.dumps(d)
+
+    def action(self, level, what, action, who):
+        if not self._should_audit(level):
+            return
+        if isinstance(what, entities.Entity):
+            return
+        ships = self._format_container_dict(what)
+
+        args_ls = self._args.format(action=action,
+                                    what=what, who=who).split()
+        args_ls.append(ships)
+        cmd = [self._script]
+        cmd.extend(args_ls)
```

Comment:
Instead of building two distinct list just to merge them together:

```python
cmd = [self._script] + \
    self._args.format(action=action, what=what, who=who).split() + \
    [ships]
subprocess.check_call(cmd)
```

Also, did stdin not work again?

```python
process = subprocess.Popen(cmd, stdin=subprocess.PIPE)
output = process.communicate(input=ships)
if process.return_code != 0:
    raise subprocess.CalledProcessError(process.return_code, cmd, output)
```